includes/class-wc-gateway-stripe.php

```diff
@@ -1191,4 +1061,204 @@ public function retry_after_error( $response, $order, $retry, $force_save_source
 
         return $this->process_payment( $order->get_id(), true, $force_save_source, $response->error, $previous_error );
 	}
+    /**
+     * Create a new PaymentIntent and attempt to confirm it.
+     *
+     * @param WC_Order $order           The order that is being paid for.
+     * @param object   $prepared_source The source that is used for the payment.
+     * @return object                   An intent (that is either successful or requires an action) or an error.
+     */
+    public function create_and_confirm_intent( $order, $prepared_source ) {
+        // The request for a charge contains metadata for the intent.
+        $full_request = $this->generate_payment_request( $order, $prepared_source );
+
+        $request = array(
+            'source'               => $prepared_source->source,
+            'amount'               => WC_Stripe_Helper::get_stripe_amount( $order->get_total() ),
+            'currency'             => get_woocommerce_currency(),
```

Comment:
Maybe use WC_Order get_currency instead

Reply:
When the line was initially written there was no available order. Now there is one, so yes: done.